week01/assignment.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func isOperator(text string) (op string, err error) {
+	switch text {
+	case "/", "*", "+", "-":
+		op = text
+		return op, nil
+	}
+	return op, errors.New(fmt.Sprintf("Invalid operator: '%s'", text))
+}
+
+func isFloat(text string) (value float64, err error) {
+	value, err = strconv.ParseFloat(text, 64)
+	return
+}
+
+func validate(text string) (a float64, b float64, op string, err error) {
+	words := strings.Fields(text)
+	if len(words) != 3 {
+		err = errors.New(fmt.Sprintf("Invalid input: '%s'", text))
+		return
+	}
+	a, err = isFloat(words[0])
+	if err != nil {
+		return
+	}
+	op, err = isOperator(words[1])
+	if err != nil {
+		return
+	}
+	b, err = isFloat(words[2])
+	if err != nil {
+		return
+	}
+	return
+}
+
+func formatFloat(f float64) string {
+	return strconv.FormatFloat(f, 'f', -1, 64)
+}
+
+func eval(text string) (err error) {
+	a, b, op, err := validate(text)
+	if err != nil {
+		return err
+	}
+	result := 0.0
+	switch op {
+	case "+":
+		result = a + b
+	case "-":
+		result = a - b
+	case "*":
+		result = a * b
+	case "/":
+		if b == 0 {
+			return errors.New("Cannot divide by zero")
+		}
+		result = a / b
+	}
+	fmt.Printf("%d %s %d = %s",
```

Comment:
You can use `%v` for the last one:

```
%d %s %d = %v
```

With this change, you don't need to convert your `result` to a string (don't need `formatFloat` function).